src/Rules/PropertyConditions/Relationships.cs

```diff
@@ -158,16 +158,16 @@ public static Condition AllAncestors(Condition c)
 
         private static bool HasSameTypeAsReferenceElement(IA11yElement e)
         {
-            if (e == null) throw new ArgumentException(nameof(e));
-            if (!Condition.Context.IsValueCreated) throw new Exception("Expected Condition.Context to be valid");
+            if (e == null) throw new ArgumentNullException(nameof(e));
+            if (!Condition.Context.IsValueCreated) throw new InvalidCastException(ErrorMessages.ExpectedValidConditionContext);
```

Comment:
InvalidCastException seems like an odd choice here... but I don't necessarily know what would be a better option

Reply:
It was an intellisense/screen reader anomaly. Fixed.